addons/hr_holidays/models/hr_leave.py

```diff
@@ -235,6 +257,19 @@ def unlink(self):
     def copy_data(self, default=None):
         raise UserError(_('A leave cannot be duplicated.'))
 
+    @api.model
+    def default_get(self, fields_list):
+        defaults = super(HolidaysRequest, self).default_get(fields_list)
+
+        LeaveType = self.env['hr.leave.type'].with_context(employee_id=defaults.get('employee_id'), leave_date=defaults.get('date_from'))
```

Comment:
Can't we simplify a bit the code and rely on the default computation for holiday_status_id ? For example using default_date_from from context in _compute_valid and _search_valid so that we don't have to add code as it should be automatically managed.

Reply:
To expand a bit, we could try to improve default_get

1/ pre-compute defaults for employee_id and date_from
-- employee_id: default_employee_id in context or current user's employee
-- date_from: default_date_from from context
--> we could for example call super().default_get(fields=['employee_id', 'date_from']) to use the ORM default_get in a clean way, notably using user-defined default values and re-use thos values to update the context before re-calling super().default_get(fields=fields)

2/ call super with context having default_employee_id and default_date_from

--> it will notably call the default for holiday_status_id
--> it performs a search on holiday type. _search_valid will use default_date_from and we could update the _search to use default_employee_id from context instead of an ugly 'employee_id'

3/ All glory. Normally it should have found a default status correctly computed